### PR TITLE
Add support for dynamicallly loading profiles from configmaps

### PR DESCRIPTION
**Pull request by accorvin**
This adds support to the profiles code to dynamically identify any
config maps matching a target label and load profile data from matched
profiles.

Profiles will be processed alphabetically by name, providing for a level
of determinism on how the final resulting profile will be constructed.

### CODE REVIEW

**Comment by accorvin** [comment]
@vpavlin this is a first try at what we discussed to allow loading singleuser profiles from arbitrary configmaps that match a label selector.

I haven't tested this yet (I'm working on doing so), but I wanted to get your thoughts on this before I proceed further.

File: jupyterhub_singleuser_profiles/profiles.py [hunk 1]
```diff
@@ -87,12 +99,15 @@ def load_profiles(self, secret_name="jupyter-singleuser-profiles", filename=None
     self.profiles = []
     self.sizes = []
     if self.api_client:
-      config_map_yaml = self.read_config_map(secret_name, key_name)
-      if config_map_yaml:
-        self.sizes.extend(config_map_yaml.get("sizes", [self.empty_profile()]))
-        self.profiles.extend(config_map_yaml.get("profiles", [self.empty_profile()]))
-      else:
-        _LOGGER.error("Could not find config map %s" % config_map_name)
+      profiles_config_maps = ['secret_name']
```
Comment (vpavlin) [inline jupyterhub_singleuser_profiles/profiles.py]: This should be a variable `secret_name` not a string

**Comment by vpavlin** [comment]
BTW we will need to update README for this change as well.

And as I don't have a way to document this anywhere, the simplest way I have is:

1. Create a new branch in your fork of jupyterhub-ocp-oauth.
2. Change https://github.com/AICoE/jupyterhub-ocp-oauth/blob/master/requirements.txt#L3 to install from your repo and branch
3. Remove `registry` and `repository` from ODH CR so that it uses builds instead of pulling from quay.
4. After you deploy ODH, change the build `jupyterhub-img` to pull from your repo fork and branch

That way you will have your changes in deployed Jupyterhub image. It might be good to scale down the ODH operator after successful deployment to prevent the rewrites of the BC

Makes sense?

**Comment by vpavlin** [comment]
I wanted to try it out:) So here is a commit which fixes issues with the PR: https://github.com/vpavlin/jupyterhub-singleuser-profiles/commit/395187cd777afb5262b2470c2d2841359cd4d169

Would be great if you can pull in the changes and add some documentation

Otherwise works perfectly! (I am using this change in MOC right now, to provide bigger size to some students:) )

File: jupyterhub_singleuser_profiles/profiles.py (hunk 1 above)
Comment (accorvin) [reply jupyterhub_singleuser_profiles/profiles.py]: Good catch. This was stupid lol. I'll pull in your commit fixing this.

**Comment by accorvin** [comment]
@vpavlin I've added a commit updating the README with information about this new feature. Let me know if you want more detailed instructions on usage.

**Comment by vpavlin** [comment]
@accorvin Works fine for me. Thanks!

Once this is merged I'll also add some example yaml files.

We'll need to update the https://github.com/AICoE/jupyterhub-ocp-oauth/blob/master/requirements.txt#L3 once this is merged with the commit id. 

@LaVLaS any comments?

**Pull request merged**